src/js/engagement_view/src/services/fetch.ts

```diff
@@ -14,7 +14,7 @@ export const apiFetch = async (urlSlug: string, method = "GET") => {
 };
 
 
-export const apiFetchWithBody = async (urlSlug: string, method = "POST", body: string) => {
+export const apiPostRequestWithBody = async (urlSlug: string, method = "POST", body: string) => {

```

Comment:
Everywhere you call apiPostRequestWithBody you specify "post"; why not just remove this param? It seems encoded in the method name now.

Reply:
Good idea, I think the rationale was to use the param in case we ever wanted to call a different method, but you're totally right, this doesn't make sense for the name - especially when we only support GET + POST.